symbols/ram.yml

```diff
@@ -18,6 +18,41 @@ ram:
     More specifically, this file is a dumping ground for addresses that are useful to know about, but don't fall in the address ranges of any of the other files. Dynamically loaded constructs that do fall within the address range of a relevant binary should be listed in the corresponding YAML file of that binary, since it still has direct utility when reverse-engineering that particular binary.
   functions: []
   data:
+    - name: MEMORY_ALLOCATION_TABLE
```

Comment:
> About [aed4eaf](https://github.com/UsernameFodder/pmdsky-debug/commit/aed4eafc41a628185ba162c7531c3cab8819dcdd): Those symbols were previously defined in arm9.yml since they are placed in the area that used to contain the code moved to the ITCM at runtime. They can't be defined there since that triggers an overlap error with the ITCM region.

Gah, that's unfortunate. I think I would prefer if we just move them into `itcm.yml` and not list addresses for the `-ITCM` versions? We can put "[runtime]" at the start of the descriptions (like we do with some other symbols) or something. The normal versions still correspond to `arm9.bin`, so it conceptually makes some sort of sense...

Keeping them under `arm9.yml` is useful when loading stuff into Ghidra so that these functions are properly labeled. If we put them in `ram.yml` then it'll be more annoying to import them into Ghidra (especially with the NTRGhidra stuff which is fully automated).

Reply:
I think you're getting a bit confused about these symbols. They don't exist in the binaries. Once the game is started, the following steps happen:

1. The code located at [EU]0x20B3CC0 gets moved to the ITCM (0x1FF8000)
2. The area that used to contain that code [EU]0x20B3CC0 gets filled with zeroes
3. That area is now free, so it's used to store the memory table and related symbols

So these symbols only exist in the RAM, so it makes sense to have them declared in ram.yml.

Reply:
I didn't explain myself well. Your steps are correct, but even then, there's still a big advantage to putting them in `arm9.yml` rather than `ram.yml`. Other functions in `arm9.bin` (in particular those related to the memory allocator) refer to these runtime-only symbols. If they aren't listed as part of arm9, they won't be imported into Ghidra, and Ghidra won't be able to resolve those accesses into readable names, and you won't be able to create appropriate global structs at those addresses to help the decompiler.

Note the second paragraph in the description for `ram.yml`:

> More specifically, this file is a dumping ground for addresses that are useful to know about, but don't fall in the address ranges of any of the other files. Dynamically loaded constructs that do fall within the address range of a relevant binary should be listed in the corresponding YAML file of that binary, since it still has direct utility when reverse-engineering that particular binary.

That's why some other runtime-only symbols are not listed in `ram.yml`, but rather in the normal files and annotated with "[Runtime]" in the description.

Reply:
Oh, I see. That makes sense. I'll undo https://github.com/UsernameFodder/pmdsky-debug/commit/aed4eafc41a628185ba162c7531c3cab8819dcdd then.
But we still need a way to skip the overlap check for these symbols.

Reply:
That seems too hacky I think. These symbols have nothing to do with the ITCM, since they aren't part of the code that gets moved there nor are ever loaded into the ITCM themselves.

Reply:
It doesn't seem all that bad to me. It has to do with the ITCM because it gets written to the section of ARM9 that used to contain ITCM stuff. That checks out given that blocks/symbols/subregions are all organized in terms of address ranges.

The only downside I can think of is clarity, since programmatically, putting a symbol in a subregion is no different from putting it directly in the main block. So I'm not sure it's worth modifying the format for a minor issue of clarity for a small collection of symbols caused by ITCM weirdness that only really happens in one place. That's why I think it would be simpler to just defer modifying the format until this actually becomes a functional problem.

Reply:
I never liked that philosophy because this kind of problems are usually easier to deal with when they first show up rather than when you reach a point where you're forced to properly address them, but oh well. I guess we can hope it doesn't become a larger problem or that it's easily fixable in the future if it does.

Reply:
I would agree with you in some situations, but in this particular case I can't envision this becoming a larger problem that isn't just as easy to fix later. It's most likely going to be just moving around some symbols and messing with the checks somehow.